### PR TITLE
Refactors the atmos watertank backpack

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Refactors the atmos firefighting backpack to be a bit more sane, and some other slightly related code to use the `TRUE/FALSE` defines

## Why It's Good For The Game
Code following guidelines good

## Testing
Tested if the metal foam still worked correctly, the nanofrost still moved and welded vents, water still did water things and if it could refill in every mode

## Changelog
NPFC

### CODE REVIEW

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 1]
```diff
@@ -288,42 +287,40 @@
...
-			return
-		if(metal_synthesis_cooldown < 5)
-			var/obj/effect/particle_effect/foam/F = new /obj/effect/particle_effect/foam(get_turf(target), 1)
+
+	switch(nozzle_mode)
+		if(NANOFROST)
+			if(Adj)
+				return //Safety check so you don't blast yourself trying to refill your tank
+			if(reagents.total_volume < 100)
+				to_chat(user, "You need at least 100 units of water to use the nanofrost launcher!")
```
Comment (Henri215) [inline code/game/objects/items/weapons/tanks/watertank.dm:296]: These to_chats could use a span class

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 2]
```diff
@@ -288,42 +287,40 @@
...
+				return
+			if(nanofrost_cooldown)
+				to_chat(user, "Nanofrost launcher is still recharging")
```
Comment (Henri215) [inline code/game/objects/items/weapons/tanks/watertank.dm:299]:
```suggestion
				to_chat(user, "Nanofrost launcher is still recharging.")
```

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 3]
```diff
@@ -335,13 +332,10 @@
...
-	S.set_up(6, FALSE, loc, null, 1)
+	S.set_up(6, FALSE, loc, null, TRUE)
 	S.start()
-	var/obj/effect/decal/cleanable/flour/F = new /obj/effect/decal/cleanable/flour(src.loc)
-	F.color = "#B2FFFF"
-	F.name = "nanofrost residue"
-	F.desc = "Residue left behind from a nanofrost detonation. Perhaps there was a fire here?"
-	playsound(src,'sound/effects/bamf.ogg',100,1)
+	new /obj/effect/decal/cleanable/flour/nanofrost(get_turf(src))
+	playsound(src, 'sound/effects/bamf.ogg', 100, 1)
```
Comment (Henri215) [inline code/game/objects/items/weapons/tanks/watertank.dm:338]:
```suggestion
	playsound(src, 'sound/effects/bamf.ogg', 100, TRUE)
```

File: code/game/objects/items/weapons/tanks/watertank.dm (hunk 1 above)
Comment (DGamerL) [reply code/game/objects/items/weapons/tanks/watertank.dm]: Done

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 4]
```diff
@@ -288,42 +287,40 @@
...
-			spawn(100)
-				if(src)
-					metal_synthesis_cooldown--
-		else
-			to_chat(user, "Metal foam mix is still being synthesized.")
-			return
+			addtimer(CALLBACK(src, PROC_REF(cool_down)), 10 SECONDS)
+
+/obj/item/extinguisher/mini/nozzle/proc/cool_down()
+	metal_synthesis_cooldown--
```
Comment (Henri215) [inline code/game/objects/items/weapons/tanks/watertank.dm:323]: cool down or cooldown here? And could maybe add "synthesis" to the name too.

Comment (DGamerL) [reply code/game/objects/items/weapons/tanks/watertank.dm]: Renamed it to `metal_cooldown`

**Review by Henri215 — APPROVED** [review]
Lewcing good to me

**Review by S34NW — APPROVED** [review]

**Pull request merged**